tools/java/org.hl7.fhir.tools.core/src/org/hl7/fhir/tools/implementations/ember/EmberModel.java

```diff
@@ -0,0 +1,359 @@
+                } else {
+                    if (typeRef.isElementReference()) {
+                        elementType = getComponentStructName(elementType);
+                    }
+                    block.ln(getFieldDefinition(fieldName, m.typify(elementType)));
+                }
+            }
+        }
+    }
+
+    private static final class Modifier {
+        private final boolean isSlice;
+
+        public Modifier(ElementDefn el) {
+            this.isSlice = el.getMaxCardinality() == null || el.getMaxCardinality() > 1;
+        }
+
+        public String typify(String type) {
+            type = dasherize(type);
+            String result = type;
+            if (isSlice && ! "string".equals(type) && ! "interface{}".equals(type) && ! "number".equals(type) && !"date".equals(type) && !"boolean".equals(type)) {
+                result = "DS.hasMany('".concat(type).concat("', {embedded: true})");
+            } else if (! "string".equals(type) && ! "interface{}".equals(type) && ! "number".equals(type) && !"date".equals(type) && !"boolean".equals(type)) {
+                result = "DS.belongsTo('".concat(type).concat("', {embedded: true})");
+            } else {
+              result = "DS.attr('" + type + "')";
+            }
+            return result;
+        }
+    }
+
+    private String getFieldName(ElementDefn elementDefinition, TypeRef type) {
+        StringBuilder elementName = new StringBuilder(elementDefinition.getName().replace("[x]", ""));
+        if (elementDefinition.getTypes().size() > 1) {
+            elementName.append(capitalize(type.getName()));
+        }
+        if ("data".equals(elementName.toString())) {
+          return "foo";
+        }
+        return capitalize(elementName.toString());
+    }
+
+    private String getFieldDefinition(String fieldName, String fieldTypeModifier, String fieldType) {
+        String typeName = lowercase(fieldName);
+        fieldName = lowercase(fieldName);
+        return String.format("%s:  %s('%s', {embedded: true})", fieldName, fieldTypeModifier, fieldType, typeName, typeName);
+    }
+
+    private String getFieldDefinition(String fieldName, String fieldType) {
+        String typeName = lowercase(fieldName);
+        return String.format("%s: %s", typeName, fieldType, typeName, typeName);
+    }
+
+    private boolean isComponent(ElementDefn elementDefinition) {
+        return elementDefinition.getTypes().isEmpty();
+    }
+
+    private String getComponentStructName(ElementDefn elementDefinition) {
+        return getComponentStructName(elementDefinition.getDeclaredTypeName());
+    }
+
+    private String getComponentStructName(String typeName) {
+        typeName = capitalize(typeName);
+        if (! typeName.startsWith(name)) {
+            typeName = name.concat(typeName);
+        }
+        return typeName;
+    }
+
+    private void generateComponentStruct(GenBlock block, ElementDefn elementDefinition) throws Exception {
```

Comment:
Since you're creating your own `componentBlock` and writing to a new file, you can remove the `block` argument from this function signature.  It's dead code.